app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -17,12 +17,24 @@ def parse
 
   def networks
     collector.networks.each do |network|
-      persister_switches = id_of_external_network?(network.id) ? persister.external_distributed_virtual_switches : persister.distributed_virtual_switches
+      is_external = id_of_external_network?(network.id)
+
+      persister_switches = is_external ? persister.external_distributed_virtual_switches : persister.distributed_virtual_switches
+      attrs_to_assign = {
+        :name    => network.name,
+        :uid_ems => network.id
+      }
+
+      if is_external
+        datacenter = network.data_center
+        ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(datacenter.href)
+        parent_datacenter = persister.datacenters.lazy_find(ems_ref)
+
+        attrs_to_assign[:parent] = parent_datacenter if id_of_external_network?(network.id)
```

Comment:
Minor cleanup, if we are in `is_external` then `if id_of_external_network?(network.id)` will always be true

Reply:
Oh yes, sorry I left it behind... I'll fix it in a new cleanup review